core/src/test/java/org/elasticsearch/search/sort/ScoreSortBuilderTests.java

```diff
@@ -0,0 +1,80 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.search.sort;
+
+
+import org.elasticsearch.common.ParseFieldMatcher;
+import org.elasticsearch.common.settings.Settings;
+import org.elasticsearch.common.xcontent.XContentFactory;
+import org.elasticsearch.common.xcontent.XContentParser;
+import org.elasticsearch.index.query.QueryParseContext;
+
+import java.io.IOException;
+
+public class ScoreSortBuilderTests extends AbstractSortTestCase<ScoreSortBuilder> {
+
+    @Override
+    protected ScoreSortBuilder createTestItem() {
+        return new ScoreSortBuilder().order(randomBoolean() ? SortOrder.ASC : SortOrder.DESC);
+    }
+
+    @Override
+    protected ScoreSortBuilder mutate(ScoreSortBuilder original) throws IOException {
+        ScoreSortBuilder result = new ScoreSortBuilder();
+        if (original.order() == SortOrder.ASC) {
+            result.order(SortOrder.DESC);
+        } else {
+            result.order(SortOrder.ASC);
+        }
+        return result;
+    }
+
+    /**
+     * test passing null to {@link ScoreSortBuilder#order(SortOrder)} is illegal
+     */
+    public void testIllegalOrder()  {
+        try {
+            new ScoreSortBuilder().order(null);
+            fail("NPE expected");
+        } catch (NullPointerException e) {
```

Comment:
Looks great, I just tried it and it works for us. Will use it more often I think, better than `@Test(expected = ...)` which I think is a forbidden API now.